src/extlibs/hrvo/simulator.h

```diff
@@ -275,158 +245,57 @@ class Simulator
-     *
-     * The preferred speed of an agent is the speed it would choose
-     *             to take if it were not influenced by other agents.
-     *
-     * @param agentNo    The number of the agent whose preferred speed is to be modified.
-     * @param prefSpeed  The replacement preferred speed.
-     */
-    void setAgentPrefSpeed(std::size_t agentNo, float prefSpeed);
+    // List of agents (robots) in this simulation
+    std::vector<std::unique_ptr<Agent>> agents_;
+    // TODO (#2373): Remove goals_ list when goal is a part of Agent
+    std::vector<std::unique_ptr<Goal>> goals_;
 
-    /**
-     *      Sets the radius of a specified agent.
-     *
-     * @param agentNo  The number of the agent whose radius is to be modified.
-     * @param radius   The replacement radius.
-     */
-    void setAgentRadius(std::size_t agentNo, float radius);
+   private:
+    // friendly robot id to agent index
+    std::map<unsigned int, unsigned int> friendly_robot_id_map;
 
-    /**
-     *      Sets the "uncertainty offset" of a specified agent.
-     *
-     * The uncertainty offset is the amount velocity obstacles ar
-     *             widened to allow for uncertainty in the position and velocity
-     *             of a differential drive agent.
-     *
-     * @param agentNo            The number of the agent whose uncertainty offset is to be
-     * modified.
-     * @param uncertaintyOffset  The replacement uncertainty offset.
-     */
-    void setAgentUncertaintyOffset(std::size_t agentNo, float uncertaintyOffset);
+    // PrimitiveSet which includes the path which each friendly robot should take
+    TbotsProto::PrimitiveSet primitive_set_;
 
-    /**
-     *      Sets the velocity of a specified agent.
-     *
-     * @param agentNo   The number of the agent whose velocity is to be modified.
-     * @param velocity  The replacement velocity.
-     */
-    void setAgentVelocity(std::size_t agentNo, const Vector2 &velocity);
+    // True if the ball should be treated as an agent (obstacle)
+    // NOTE: This will take effect the next time we receive a world, and we know
+    //       the current ball position and velocity
+    bool add_ball_agent = false;
 
-    /**
-     *      Sets the time step of the simulation.
-     *
-     * @param timeStep  The replacement time step of the simulation.
-     */
-    void setTimeStep(float timeStep)
-    {
-        timeStep_ = timeStep;
-    }
+    // The scale which friendly robots should be larger than friendly robots
+    // This scale is used to avoid close encounters, and reduce chance of collision
+    const float friendly_robot_radius_scale = 1.25f;
 
-    //	private:
-   public:
-    Simulator(const Simulator &other);
-    Simulator &operator=(const Simulator &other);
+    // The scale which enemy robots should be larger than their actual size
+    // This scale is used to avoid close encounters, and reduce chance of collision
+    const float enemy_robot_radius_scale = 2.f;
```

Comment:
can we get this from RobotNavigationObstacleConfig

Reply:
Honestly, I don't know how we're suppose to use the yaml configs. I'm also not sure what benefit it would give us to put the constants in a separate file. Would the goal be to be able to dynamically tune the values? 
For now I updated the constants to be `static constexpr`, similar to the constants in `theta_star_path_planner.h`.

Reply:
it'd be nice to be dynamic but this could be a later ticket